libs/leonsans/src/core/model.ts

```diff
@@ -0,0 +1,282 @@
+  drawing_: Drawing[];
+  data_: ModelData[];
+  rect_: Rect;
+  align_: Align;
+  scale_: number;
+  fontRatio_: number;
+
+  constructor() {
+    this.lineWidth_ = 1;
+    this.drawing_ = [];
+    this.data_ = [];
+    this.rect_ = {
+      x: 0,
+      y: 0,
+      w: 0,
+      h: 0,
+    };
+    this.align_ = 'left';
+    this.scale_ = 1;
+    this.fontRatio_ = 1;
+  }
+
+  get data() {
+    return this.data_;
+  }
+
+  get lineWidth() {
+    return this.lineWidth_;
+  }
+
+  get fontRatio() {
+    return this.fontRatio_;
+  }
+
+  get scale() {
+    return this.scale_;
+  }
+
+  get rect() {
+    return this.rect_;
+  }
+
+  get drawing() {
+    return this.drawing_;
+  }
+
+  set align(v) {
+    if (this.align_ != v) {
+      this.align_ = v;
+      this.setPosition();
+    }
+  }
+
+  get align() {
+    return this.align_;
+  }
+
+  /**
+   * 리온 산스의 위치를 설정합니다.
+   * @param x x 좌표
+   * @param y y 좌표
+   */
+  position(x: number, y: number) {
+    if (this.rect_.x != x || this.rect_.y != y) {
+      this.rect_.x = x;
+      this.rect_.y = y;
+      this.setPosition();
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * Set position of each typo
+   */
+  setPosition() {
+    for (const d of this.data_) {
+      d.rect.x =
+        d.originPos.x + this.rect_.x + getAlignGapX(this.align_, d.alignGapX);
+      d.rect.y = d.originPos.y + this.rect_.y;
+    }
+  }
+
+  updateDrawingPaths() {
+    for (const d of this.data_) {
+      d.drawingPaths = addRectToPaths(getPaths(this, d, -1, false), d);
+    }
+  }
+
+  updatePatternPaths(pathGap: number) {
+    const total = this.data_.length;
+    let i, d;
+    for (i = 0; i < total; i++) {
+      d = this.data_[i];
+      d.relativePatternPaths = getPaths(this, d, pathGap, true);
+    }
+  }
+
+  updateWavePaths(pathGap: number) {
+    const total = this.data_.length;
+    let i, d;
+    for (i = 0; i < total; i++) {
+      d = this.data_[i];
+      d.relativeWavePaths = getPaths(this, d, pathGap, false);
+    }
+  }
+
+  updateGuide() {
+    for (const d of this.data_) {
+      d.guide = getGuide(d.typo, this.scale);
+      d.grid = getGrid(d.typo, this.scale);
+    }
+  }
+
+  update(
+    text: string,
+    width: number,
+    breakWord: boolean,
+    weight: number,
+    size: number,
+    tracking: number,
+    leading: number,
+  ) {
+    const fontW = getFontW(weight);
+    const weightRatio = getWeightRatio(fontW);
+    const circleRound = getCircleRound(fontW);
+    const scale = getScale(size);
+    const m_tracking = getTracking(tracking, scale);
+    const m_leading = getLeading(leading, scale);
+    const fontRatio = getFontRatio(weightRatio);
+
+    this.fontRatio_ = fontRatio;
+    this.scale_ = scale;
+    this.lineWidth_ = getLineW(fontW, scale);
+
+    const textGroup = getTextGroup(text, scale, width, breakWord);
+
+    let maxW = 0; // max width
+    let maxH = 0; // max height
+
+    let tx = 0; // total x
+    let ty = 0; // total y
```

Comment:
total x, total y라는 게 정확히 무엇인가요? 모든 typo의 width를 더한 값인가?

Reply:
```suggestion
    let totalX = 0; // total x
    let totalY = 0; // total y
```

갠적으로 w, h, x, y와 같은 누구나 알 수 있을만한 줄임말 말고는 변수명을 바꿔두는 게 어떨까 합니다

Reply:
이것도 원래 이렇게 작명되어있었던 거라... 옆에 작성해둔 주석도 제가 임의로 추측해서 적어둔 거긴 한데 일단 저렇게 바꿔두겠습니다.